src/main/java/org/spongepowered/api/asset/Asset.java

```diff
@@ -77,6 +77,33 @@ default void copyToFile(Path output) throws IOException {
     }
 
     /**
+     * Copies this Asset to the specified 'outputDirectory' {@Link Path}.
+     *
+     * @param outputDirectory The directory to copy to
+     * @throws IOException
+     */
+    default void copyToDirectory(Path outputDirectory) throws IOException {
+        checkNotNull(outputDirectory, "outputDirectory");
+        copyToFile(outputDirectory.resolve(this.getFileName()));
```

Comment:
In my opinion, if I call `copyToDirectory` the asset should be copied to the specified directory, and not to a sub-directory of it. The other functionality can be still added later or implemented separately in plugins.

For example we could add a separate `getPath()` method that returns a string with the relative path from the asset directory to the asset itself, e.g. `plugin.jar!/assets/plugin/textures/test.png` would be `textures/test.png`